src/imitation/policies/trainer.py

```diff
@@ -45,6 +46,42 @@ def train(self, steps: int, **kwargs):
                 the training procedure.
         """
 
+    def set_logger(self, custom_logger: Optional[logger.HierarchicalLogger]):
+        """Set the internal logger.
+        For subclasses, this should also set the logger of any wrapped modules
+        (such as SB3 algorithms).
+
+        Args:
+            logger: the HierarchicalLogger instance to use.
+        """
+        self._logger = custom_logger or logger.configure()
+
+
+class TrajectoryDataset(TrajectoryGenerator):
+    def __init__(self, path: types.AnyPath, seed: int = 0):
```

Comment:
You should call superclass `__init__`. If that does something you don't want (e.g. the logger), then consider moving that logic out of the superclass.

Reply:
Done. We don't need the logger here for now but I think we decided to just make sure a logger always exists so we can set it in `PreferenceComparisons`